cmd/query/app/querysvc/query_service_v2.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package querysvc
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc/adjuster"
+	"github.com/jaegertracing/jaeger/internal/jptrace"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/iter"
+	"github.com/jaegertracing/jaeger/storage_v2/depstore"
+	"github.com/jaegertracing/jaeger/storage_v2/tracestore"
+)
+
+var ErrNoArchiveSpanStorage = errors.New("archive span storage was not configured")
+
+const (
+	defaultMaxClockSkewAdjust = time.Second
+)
+
+// QueryServiceOptionsV2 holds the configuration options for the V2 QueryService.
+type QueryServiceOptionsV2 struct {
+	// ArchiveTraceReader is used to read archived traces from the storage.
+	ArchiveTraceReader tracestore.Reader
+	// ArchiveTraceWriter is used to write traces to the archive storage.
+	ArchiveTraceWriter tracestore.Writer
+	// Adjuster is used to adjust traces before they are returned to the client.
+	// If not set, the default adjuster will be used.
+	Adjuster adjuster.Adjuster
+}
+
+// StorageCapabilities is a feature flag for query service
+type StorageCapabilities struct {
+	ArchiveStorage bool `json:"archiveStorage"`
+	// TODO: Maybe add metrics Storage here
+	// SupportRegex     bool
+	// SupportTagFilter bool
```

Comment:
@yurishkuro do you have context behind these items? do we still want to do them?